docs/developer/decision-records/2025-05-28-dataspace-context/README.md

```diff
@@ -0,0 +1,41 @@
+# Dataspace Context
+
+## Decision
+
+We'll introduce the concept of "Dataspace context", that will permit a connector to being able to interact with multiple
+dataspaces with potential different protocol versions, authentication and profiles 
+
+## Rationale
+
+At the moment the EDC connector only permits to support different protocol version at the same time, by having pluggable
+modules that add support for them.
+
+A Dataspace context is a more hi level concept that enables to define a set of components that will permit the connector
+to interact also with different dataspaces, by defining a set of these capabilities:
+- wire protocol, version and binding
+- authentication protocol
+- identifier
+- scopes
+- ...
+
+## Approach
+
+A Dataspace context will be bound to a single DSP endpoint.
+
+Please note that will be possible to define multiple contexts for a single protocol version, there will be an api context
+for every dataspace.
+
+By default, there will be a single context registered for every DSP module (v0.8, 2024/1, 2025/1, ...) with default identity
+service (DCP), default identifier (DID) and so on.
+
+The adopter will then be able to override these ones with dataspace specific contexts with potentially different capabilities.
+That could be done through configuration, under the `edc.dataspace.context` group.
+If no setting is found, the current default `protocol` context will be used, otherwise:
+
```

Comment:
Can you provide some examples? I'm not quite clear about this. For example, are you saying there will be "default" contexts for each DSP module that act as a fallback if a context is not explicitly configured?

Reply:
I'd like to get a better handle on this and document it as a strategy before we get too far into the implementation, or at least have a spike to clarify the DR first. Should we discuss during the weekly?

Reply:
Agreed, let's discuss further. I think we need to walk through the scenarios.